File: C++ Solutions/MaximumDepthOfBinaryTree104.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};
int maxDepth(TreeNode *root)
{
    if (root == NULL)
        return 0;
    return 1 + max(maxDepth(root->left), maxDepth(root->right));
}
int main()
{
    TreeNode *root = new TreeNode(3);
    root->left = new TreeNode(9);
    root->left->left = new TreeNode(15);
    root->left->right = new TreeNode(7);
    root->right = new TreeNode(20);
    cout << maxDepth(root);
    return 0;
}

File: JAVA Solutions/MaximumDepthOfBinaryTree104.java
public class MaximumDepthOfBinaryTree104 {
    public static void main(String[] args) {
        MaximumDepthOfBinaryTree104 s = new MaximumDepthOfBinaryTree104();
        TreeNode root = new TreeNode(3);
        root.left = new TreeNode(9);
        root.left.left = new TreeNode(15);
        root.left.right = new TreeNode(7);
        root.right = new TreeNode(20);
        System.out.println(s.maxDepth(root));
    }
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;
        int leftDepth = 1 + maxDepth(root.left);
        int rightDepth = 1 + maxDepth(root.right);
        return Math.max(leftDepth, rightDepth);
    }
}